src/fhirtypes/ElementDefinition.ts

```diff
@@ -1300,6 +1311,33 @@ export class ElementDefinition {
     }
   }
 
+  private checkChildFixedValue(child: ElementDefinition, fhirValue: any) {
+    const childType = child.type[0].code;
+    const fixedX = `fixed${upperFirst(childType)}` as keyof ElementDefinition;
+    const patternX = `pattern${upperFirst(childType)}` as keyof ElementDefinition;
+    const currentChildValue = child[fixedX] ?? child[patternX];
+    if (currentChildValue != null) {
+      // find the element on fhirValue that would get assigned to the child
+      const childPath = child.id.replace(`${this.id}.`, '').split('.');
+      let newChildValue = fhirValue;
+      for (const pathPart of childPath) {
+        if (newChildValue != null) {
+          newChildValue = newChildValue[pathPart];
+        }
```

Comment:
I think this causes an issue in the case of an array. Consider this example:
```
Profile: MyObs
Parent: Observation
* valueCodeableConcept.coding.code = #bar
* valueCodeableConcept = #foo
```
This is pretty much the same situation as your test on `valueQuantity`, but because `valueCodeableConcept` contains an array, the test does not seem to work. You end up with `patternCode: "bar"` on `valueCodeableConcept.coding.code`, and `patternCodeableConcept: {coding: [{code: "foo"}]}` on `valueCodeableConcept`, and these two rules conflict.

The `fixedByAnyParent` function checks if a value is fixed by any parent of an element, and returns that element if so, and it does work with arrays, so my first thought was that you could use that function, but of course that is an issue because you can't use that function until you actually do fix the value on the parent element, but we are trying to decide if it is valid to fix the value. So one possibility would be to fix the value on the parent, determine if that fixing violated anything on the child elements, and un-fix the value on the parent if so. But maybe it is just simpler to add array handling logic here.